simulator/simulator.h

```diff
@@ -18,13 +18,39 @@ namespace et_simulator {
 class Simulator
 {
  private:
+  // Context tree
   CCTree * tree;
 
+  // Name of the files
   std::string tracefile;
   std::string namesfile;
 
-  void read_name_file(std::ifstream & name_file);
-  void read_trace_file(std::ifstream & in);
+  // Simulation values
```

Comment:
My main concern is moving these local variables of read_trace_file() to instance variables of Simulator. This broadened the scope of the variables, and IMO makes the code less readable and more difficult to debug/test. Variables in general should be declared in the narrowest scope possible (https://google.github.io/styleguide/cppguide.html#Local_Variables).

A better way is to declare them inside each case statement, and explicitly pass the data dependence through parameters, like this:

case 'A': {
     int object_id, size, thread_id, method_id;
     std::string type;
     if(!parse_object_allocation(trace, &object_id, &type, &size, &thread_id, &method_id)) {
           // Log bad allocation entry
          return false;
      }
      tree->handle_object_allocation(object_id, size, type, thread_id, method_id);
      return true;
}

However, since now each case statement always returns true/false, you can probably make parse_object_allocation() directly call handle_object_allocation(), and declare these as local variables inside parse_object_allocation().